crates/proto/src/xfer/dns_response.rs

```diff
@@ -263,7 +263,7 @@ impl DnsResponse {
     pub fn negative_type(&self) -> Option<NegativeType> {
         let response_code = self.response_code();
         let ttl_from_soa = self.negative_ttl();
-        let has_soa = ttl_from_soa.map_or(false, |_| true);
+        let has_soa = ttl_from_soa.is_some_and(|_| true);
```

Comment:
nit: this can be further simplified
```suggestion
        let has_soa = ttl_from_soa.is_some();
```

Reply:
Good catch!